### PR TITLE
feat(StageInstance): Add support for associated guild event

### PR DESCRIPTION
**Pull request by suneettipirneni**
**Please describe the changes this PR makes and why it should be merged:**

Adds support to associated cached scheduled events with a stage instance if applicable.

- https://github.com/discord/discord-api-docs/pull/4583

**Status and versioning classification:**
- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating
- This PR changes the library's interface (methods or parameters added)

### CODE REVIEW

File: packages/discord.js/src/structures/StageInstance.js [hunk 1]
```diff
@@ -63,6 +63,16 @@ class StageInstance extends Base {
     } else {
       this.discoverableDisabled ??= null;
     }
+
+    if ('guild_scheduled_event_id' in data) {
+      /**
+       * The associated scheduled event Id of this stage instance
```
Comment (Jiralite) [inline packages/discord.js/src/structures/StageInstance.js:69]:
```suggestion
       * The associated guild scheduled event id of this stage instance
```

File: packages/discord.js/src/structures/StageInstance.js [hunk 2]
```diff
@@ -83,6 +93,15 @@ class StageInstance extends Base {
     return this.client.guilds.resolve(this.guildId);
   }
 
+  /**
+   * The associated scheduled event of this stage instance
```
Comment (Jiralite) [inline packages/discord.js/src/structures/StageInstance.js:97]:
```suggestion
   * The associated guild scheduled event of this stage instance
```

File: packages/discord.js/src/structures/StageInstance.js [hunk 3]
```diff
@@ -83,6 +93,15 @@ class StageInstance extends Base {
...
+   * @type {?GuildScheduledEvent}
+   * @readonly
+   */
+  get scheduledEvent() {
+    return this.guild?.scheduledEvents.resolve(this.guildScheduledEventId);
+  }
```
Comment (Jiralite) [inline packages/discord.js/src/structures/StageInstance.js:103]: So doesn't this possibly return `undefined`? Might want to chuck a `?? null` after it maybe?

Comment (suneettipirneni) [reply packages/discord.js/src/structures/StageInstance.js:103]: `this.guild` is `null` if not found so it should never be `undefined`.

Comment (imranbarbhuiya) [reply packages/discord.js/src/structures/StageInstance.js]: if `this.guild` is null then `this.guild?.scheduledEvents.resolve(this.guildScheduledEventId)` will return `undefined`. no?

Comment (suneettipirneni) [reply packages/discord.js/src/structures/StageInstance.js]: Yeah it's been fixed.

File: packages/discord.js/src/structures/StageInstance.js [hunk 4]
```diff
@@ -83,6 +93,15 @@ class StageInstance extends Base {
     return this.client.guilds.resolve(this.guildId);
   }
 
+  /**
+   * The associated guild scheduled event of this stage instance
+   * @type {?GuildScheduledEvent}
+   * @readonly
+   */
+  get scheduledEvent() {
```
Comment (ImRodry) [inline packages/discord.js/src/structures/StageInstance.js:101]: I think you should either call this guildScheduledEvent or call the other prop scheduledEventId so their names match

**Review by SpaceEEC — APPROVED** [review]

**Review by iCrawl — APPROVED** [review]

**Review by kyranet — APPROVED** [review]

**Comment by suneettipirneni** [comment]
Upstream PR merged

**Pull request merged**